docs/upgrade/UPGRADE-unreleased.md

```diff
@@ -6,3 +6,54 @@ This guide contains instructions to upgrade from version v7.0.0-beta5 to Unrelea
+ - property `cartItems` has been renamed to `items`
+ - method `getCartItemById` has been renamed to `getItemById`
+ - method `getQuantifiedProductsIndexedByCartItemId` has been renamed to `getQuantifiedProductsIndexedByItemId`
+ - method `findSimilarCartItemByCartItem` has been renamed to `findSimilarItemByItem`
+
+`Cart` is now entity, so change your usages appropriately:
+ - properties `user` and `cart_identifier` were moved from `CartItem` to `Cart`
+    - has new methods `getCartIdentifier`, `setModifeiedNow` and `setModifiedAt`
+ - new property `modifiedAt`
+ - method's `addProduct` first parameter `CustomerIdentifier` has been removed 
+ - method's `mergeWithCart` third parameter `CustomerIdentifier` has been removed
+ - methods `addItem`, `removeItemyId`, `getItems`, `getItemsCount`, `changeQuantities`, `getQuantifiedProductsIndexedByItemId`, `mergeWithCart` and `findSimilarItemByItem` has been changed so update them if you extended them
+ 
+`CartFactory` has been refactored, so change your usages appropriately:
+ - methods `get` and `createNewCart` has been removed use `CartFacade`'s `findCartOfCurrentCustomer`, `findCartByCustomerIdentifier`, `getCartOfCurrentCustomerCreateIfNotExists` or `getCartByCustomerIdentifierCreateIfNotExists` instead
+    - methods starting with `find` can return `null`
+    - methods ending with `CreateIfNotExists` will always create new `Cart` in database, so use this methods only in case you are adding some items
+ - method `clearCache` has been removed
+ - attributes `carts`, `cartItemRepository` and `cartWatcherFacade` has been removed
+ 
+`CartFacade` has been refactored, so change your usages appropriately:
+ - it has four new methods for working with `Cart`
+    - methods `findCartOfCurrentCustomer` and `findCartByCustomerIdentifier` will return `Cart` if it contains at least one `CartItem` or `null` if it is empty
+    - methods `getCartOfCurrentCustomerCreateIfNotExists` and `getCartByCustomerIdentifierIfNotExists` will return `Cart` if it contains at least one `CartItem` or create one if it does not
+ - methods `addProductToCart`, `changeQuantities`, `deleteCartItem`, `getQuantifiedProductsOfCurrentCustomerIndexedByCartItemId` and `deleteOldCarts` has been changed so update them if you extended them
+ - method `cleanCart` has been removed, use method `deleteCart` instead
+ 
+`CartMigrationFacade` has been refactored, so change your usages appropriately:
+ - it no longer contains attribute `cartFactory` attribute it now includes `cartFacade` instead
+ - all methods has been changed, so update them if you extended them
+ 
+ `CartItemRepository` class has been removed all its logic has been moved to new `CartRepository`
+ `CartItemFactory::create` and `CartItemFactoryInterface::create` methods has changed, its first parameter is no longer `CustomerIdentifier` but `Cart`
+ `CartController` on frontend has methods `indexAction` and `boxAction` updated
```

Comment:
this should not be here - it is in project-base so the users should update the usages by following all the other instructions

Reply:
removed